### PR TITLE
chore: enable codecov

### PR DESCRIPTION
**Pull request by huntye1**
## Summary by CodeRabbit

- **New Features**
	- Enhanced CI process with coverage reporting for improved testing.
	- Introduced a new configuration file for Codecov to manage coverage reporting and component statuses.
	- Added badges for code coverage and CI status in the README for better visibility.

- **Chores**
	- Updated CI workflow to include a step for uploading coverage reports.
	- Improved Codecov configuration for detailed coverage reporting and component management.

### CODE REVIEW

File: codecov.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# valid this file: curl --data-binary @codecov.yml https://codecov.io/validate
+# reference doc: https://docs.codecov.com/docs/common-recipe-list#set-project-coverage-checks-on-a-pull-request
+
+# codecov:
+#   require_ci_to_pass: false
+#   notify:
+#     wait_for_ci: false
+
```
[the remark this answers is not in the document]
Comment (huntye1) [reply codecov.yml:8]: For stability,  use default setting first.

File: codecov.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    statuses:
+      - type: project # in this case every component that doens't have a status defined will have a project type one
+        target: auto
+        threshold: 1%
+      - type: patch
+        target: auto
+        threshold: 1%
+
+  individual_components:
+    - component_id: coze-js # this is an identifier that should not be changed
```
Comment (Tecvan-fe) [inline codecov.yml:40]: 这个节点不用加 paths 吗？

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -38,8 +38,17 @@ jobs:
...
 
-      - name: Test
-        run: node common/scripts/install-run-rush.js test --verbose
+      - name: Test:cov
+        run: node common/scripts/install-run-rush.js test:cov --verbose
+
+      - name: Upload coverage reports
+        uses: codecov/codecov-action@v4
+        with:
+          files: packages/coze-js/coverage/coverage-final.json,packages/realtime-api/coverage/coverage-final.json
```
Comment (Tecvan-fe) [inline .github/workflows/ci.yml:47]: 这个支持 glob 不

File: codecov.yml (hunk 2 above)
Comment (huntye1) [reply codecov.yml:40]: 应该是刚刚误删了，我重新加上。

File: codecov.yml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+      # set component statuses here
+      # statuses:
+      #   - type: project
+      #     target: auto
+      #     threshold: 1%
+
+    - component_id: realtime-api # this is an identifier that should not be changed
+      name: realtime-api # this is a display name, and can be changed freely
+      paths:
+        - packages/realtime-api/**
```
[the remark this answers is not in the document]
Comment (huntye1) [reply codecov.yml:51]: 这个会结合report的文件一起判断的，目前来说效果应该是一样的。

**Review by Tecvan-fe — APPROVED** [review]

**Pull request merged**